### DISCUSSION TITLE
Tandoor Development

### DISCUSSION THREAD

**Discussion by flomero** [TandoorRecipes/recipes#2760] [Q&A]
After installing all the requirements (python-ldap was quite a hustle), I got to the point where `python manage.py runserver` and `yarn serve` seem to run without any errors. Unfortunately http://localhost:8080/ (django) returns a `Cannot GET /` and http://localhost:8000/ (Vue) returns `Not found`. Both are resulting from a 404 error.  What am I missing?

**Comment by smilerz** [comment]
I assume you've run the below.  what are you using for an IDE?
```
yarn build
python manage.py collectstatic
python manage.py migrate
python manage.py collectstatic_js_reverse
```

**Reply by flomero** [reply]
I'm currently using vs code and didn't run `python manage.py collectstatic` and `python manage.py collectstatic_js_reverse` before, because they aren't in the guide here: https://docs.tandoor.dev/contribute/

unfortunately, after running them now, I still have the same issue

**Reply by smilerz** [reply]
Are you running VS in WSL?

Configure your launch.json as below and you can debug in VSCode instead of manually running `runserver`.  If it still doesn't work share the output of the terminal.

``` json
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Django",
            "type": "python",
            "request": "launch",
            "program": "${workspaceFolder}/manage.py",
            "args": [
                "runserver"
            ],
            "django": true,
            "justMyCode": true
        },
```

**Reply by flomero** [reply]
i'm connecting to a raspberry pi via the remote development extension. (already checked opening the  http://localhost:8080/ and  http://localhost:8000/ directly on it with the same result)

using the debugger doesn't give me any errors:

![image](https://github.com/TandoorRecipes/recipes/assets/49640251/5ade70d5-fc8f-4c2b-b8e5-7ed9e72af30b)

**Comment by smilerz** [comment]
Not looking for errors in the debugger - stdout of the terminal

**Reply by flomero** [reply]
```
pi@flosraspi:~/Code/repos/recipes $  cd /home/pi/Code/repos/recipes ; /usr/bin/env /home/pi/Code/repos/bin/pytho.vscode-server/extensions/ms-python.python-2023.20.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 48261 -- /home/pi/Code/repos/recipes/manage.py runserver 
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
November 29, 2023 - 23:39:04
Django version 4.2.7, using settings 'recipes.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.

```

**Reply by smilerz** [reply]
If you never get any GETs in the console then this is some combination of VSCode/remote debug/django issue - you'll need to seek elsewhere for assistance.

**Reply by vabene1111** [reply]
yes I also think that smilerz is correct here @flomero the output looks correct and like the server is running, you could test it on the console of the PI itself to check if its working but it seems like a connection error.
the log on the Pi should also show connection attemts, if nothing is shown then the request isn't getting trough.

To be honest I dont think you need any remote setup, you can safely ignore ldap if you are not developing with ldap and just install it on any plattform with clone, pip install, yarn install and start up the application, shouldn't be any more than that (i always work on windows, sometimes on linux and smilerz I think on mac).

**Reply by smilerz** [reply]
I'm on Windows with WSL
